### PR TITLE
Ignore subset when pickling, detect pickled TTF (Fix)

### PR DESCRIPTION
**Pull request by moe-25**
There was a bug when a pickle created with `uni=True` was loaded with `uni=False`. The bug was obscured as the font subset was part of the pickle although this value depends on runtime parameters (e.g. `alias_nb_pages`).

Fixes:
* Detect when pickles are loaded with wrong `uni` value
* Exclude `subset` from pickling

This code demonstrates the bug (final PDF does not include page numbers):
```
pdf = fpdf.FPDF()
pdf.alias_nb_pages(None)
pdf.add_font("Roboto-Regular", fname="Roboto-Regular.ttf", uni=True)

pdf2 = fpdf.FPDF()
# re-using .pkl file generated by previous instance
pdf2.add_font("Roboto-Regular")
pdf2.set_font("Roboto-Regular", size=64)
pdf2.add_page()
pdf2.cell(txt="This is page {nb}")

pdf2.output("Test.pdf")
```

### CODE REVIEW

**Comment by Lucas-C** [comment]
Thank you for the PR !

I'll review it next week :)

@all-contributors please add @moe-25 for bug

**Comment by alexp1917** [comment]
some interesting stuff here that i think we should review before merging [here](https://github.com/PyFPDF/fpdf2/commit/f23342aa49065be80a4ce1b355761e09c48a37e2) on [this branch](https://github.com/PyFPDF/fpdf2/commits/bugfix_pickled_subset).

Not sure if this branch of code was not covered before this PR (this can be checked), but i tried to augment this test to cover it in [this commit](https://github.com/PyFPDF/fpdf2/commit/13de2856f21fbe014887d459fff3f8ff190aeb7e), and things did not go my way. first, i found that we expect `font_file_path` to be a string and not a `pathlib.Path`, then i found that we were trying to read the pickle from the original font file. then when i fixed that test, i had to add a [second commit](https://github.com/PyFPDF/fpdf2/commit/f23342aa49065be80a4ce1b355761e09c48a37e2) to make the tests pass.

the font i found is from [this site](http://www.softerviews.org/Fonts.html), direct download [here](http://www.softerviews.org/resources/Balava.7z). it's checked into git, just making sure cite the source.

happy to be convinced that this is not in fact an issue, but it looks like one to me at the moment.

**Review by alexp1917 — COMMENTED** [review]
^

**Comment by moe-25** [comment]
I tried to look at it, but failed. How did you generate the pickle? I suspect that `add_font()` was originally supposed to get the name of the .pkl file, which had to be created manually beforehand. The documentation refers to the (missing?) _Makefont_ tool for this purpose.

First, I tried with [makefont from pyfpdf](https://github.com/reingart/pyfpdf/blob/master/attic/makefont.py) but didn't get it fully working. The [PHP version's online generator](http://www.fpdf.org/makefont/) seems to extract the correct keys. However, I lacked the time to rewrite the result to python code. Not to mention understanding its generation.

I can't say what's best here: Updating the documentation, settings `uni=True` by default or dropping non-TTF-support (documented as deprecated anyway), but it's worth it's own discussion.

I came across this bug because the snipping above runs into an unnoticed trap when `font["subset"]` does not include page numbers. A rare case that could be caught nevertheless. I rebased my branch the be mergeable. This is no perfect submission but best I contribute before going to holiyday... Sorry!

File: test/fonts/test_add_font.py
Comment (Lucas-C) [inline test/fonts/test_add_font.py:64]: Could this conversion to `str` be removed?
The code should be made to support `Path`-like `fname` parameter values

File: test/fonts/test_add_font.py
Comment (Lucas-C) [inline test/fonts/test_add_font.py:85]: Same remark

File: test/fonts/test_add_font.py
Comment (Lucas-C) [inline test/fonts/test_add_font.py:103]: Nice new test! 👍

File: test/fonts/test_add_font.py
Comment (Lucas-C) [inline test/fonts/test_add_font.py:100]: Same comment about `str` conversion

File: test/fonts/test_add_font.py
Comment (Lucas-C) [inline test/fonts/test_add_font.py:133]: Same comment about `str` conversion

**Review by Lucas-C — APPROVED** [review]
Apart from a minor comment, LGTM to me, I approve this!

File: test/fonts/test_add_font.py
Comment (moe-25) [reply test/fonts/test_add_font.py:64]: Oops, sorry, I though this change was suggested by @alexp1917 ("_first, i found that we expect `font_file_path` to be a string and not a `pathlib.Path`_"). Seems to be a matter of taste. But yeah, the conversion could be removed.

Comment (Lucas-C) [reply test/fonts/test_add_font.py:64]: No, I understand it now, it's perfectly fine like this.
Merging this PR now

**Pull request merged**
